src/main/java/com/adobe/target/edge/client/service/DefaultTargetService.java

```diff
@@ -257,4 +263,20 @@ private CompletableFuture<ResponseWrapper<DeliveryResponse>> callDeliveryApiAsyn
         deliveryRequest.getDeliveryRequest(),
         DeliveryResponse.class);
   }
+
+  private void cleanUpGeoContext(TargetDeliveryRequest deliveryRequest) {
+    Geo geo = deliveryRequest.getDeliveryRequest().getContext().getGeo();
+
+    if (geo != null) {
+      if (geo.getStateCode() != null && geo.getStateCode().isEmpty()) {
```

Comment:
You could write this more succinctly as `geo.setStateCode("".equals(geo.getStateCode()) ? null : geo.getStateCode())`.  Same with city and countryCode.